### PR TITLE
#2275 Improve assertions for invalid transforms.

### PR DESCRIPTION
**Pull request by richardTingle**
These are already caught by assertions in GLRenderer, but by that time it is unclear what call pushed in the bad values

closes https://github.com/jMonkeyEngine/jmonkeyengine/issues/2275

### CODE REVIEW

**Comment by richardTingle** [comment]
closes #2275

**Comment by stephengold** [comment]
I'm generally supportive of these changes.

I recommend using the existing `Vector3f.isValidVector()` instead of defining a private method for that purpose.
Furthermore, I suggest moving `isValidQuaternion()` to `com.jme3.math.Quaternion` and making it a public method, to encourage re-use.
Finally, for `isValidNumber()` I think we should use `Float.isFinite()`, which was added in Java 1.8 .

**Comment by richardTingle** [comment]
Sure @stephengold I've made those changes

**Comment by stephengold** [comment]
I see you also added an automated test. That's great, but it needs javadoc and a copy of the JME license.

**Comment by stephengold** [comment]
Thanks for your quick responses @richardTingle. Just a few more small things ...

File: jme3-core/src/main/java/com/jme3/math/Quaternion.java [hunk 1]
```diff
@@ -1608,4 +1608,27 @@ public Quaternion clone() {
             throw new AssertionError(); // can not happen
         }
     }
+
+    /**
+     * Tests whether the argument is a valid quaternion, returning false if it's
+     * null or if any component is NaN or infinite.
+     *
+     * @param quaternion the vector to test (unaffected)
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/math/Quaternion.java:1616]: The parameter is a quaternion, not a vector.

File: jme3-core/src/test/java/com/jme3/math/QuaternionTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 jMonkeyEngine
```
Comment (stephengold) [inline jme3-core/src/test/java/com/jme3/math/QuaternionTest.java:2]: It's 2024, not 2023.

**Review by stephengold — APPROVED** [review]

File: jme3-core/src/test/java/com/jme3/math/QuaternionTest.java (hunk 2 above)
Comment (richardTingle) [reply jme3-core/src/test/java/com/jme3/math/QuaternionTest.java]: Well that's embarrassing. That isn't a copy-paste fail. I removed the year from the one I got the copyright notice from, thought "what year is it?" and wrote 2023. Anyway I've corrected that and the other issue

Comment (stephengold) [reply jme3-core/src/test/java/com/jme3/math/QuaternionTest.java]: Thanks for taking care of things.

**Pull request merged**